base_agent/loco_mc_agent.py

```diff
@@ -188,9 +188,13 @@ def init_controller(self):
         raise NotImplementedError
 
     def handle_exception(self, e):
-        logging.exception(
-            "Default handler caught exception, db_log_idx={}".format(self.memory.get_db_log_idx())
-        )
+        try:
+            err_string = "Default handler caught exception, db_log_idx={}".format(self.memory.get_db_log_idx())
```

Comment:
here's the root cause - whenever `db_log_path` is `None`, `_db_log_idx` is not initialized (https://github.com/facebookresearch/droidlet/blob/main/base_agent/sql_memory.py#L84). the [db_log_path](https://github.com/facebookresearch/droidlet/blob/main/locobot/agent/locobot_agent.py#L126) is None for the locobot agent. 

just get rid of this try catch entirely and fix the init in sql_memory.py.

Reply:
isn't this the right fix though?

`get_db_log_idx()` should raise an Exception if it doesn't exist.
`sql_memory.py` should only set a `_db_log_idx` if there is a database log. Otherwise, that attribute shouldn't be there.

Reply:
imo a class shouldn't be throwing exceptions for variables it manages internally. In the implementation currently, part of the sql_memory.py's behavior is that it throws an `AttibuteError` for getters in a certain code path (when db log in None). A cleaner behavior I think would be to just return `None` from the getter when that is the case instead of throwing an exception that the caller has to deal with.

Reply:
fair enough. Ok I went ahead and fixed it in sql_memory.py